server/s3_put_multiobject_copy_action.h

```diff
@@ -79,6 +80,10 @@ class S3PutMultipartCopyAction : public S3PutObjectActionBase {
   int part_number;
   std::string upload_id;
   size_t total_data_to_copy = 0;
+  size_t source_object_size = 0;
```

Comment:
Not sure why globals are used like this?

Reply:
As mentioned earlier, these are not global variables. These are class variables, member of `S3PutMultipartCopyAction` class and they are private by default.

Reply:
yeah, thanks for clarifying. i didnt expand the code enough above the added vars while reviewing